WordPress/Classes/ViewRelated/Reader/ReaderSeenAction.swift

```diff
@@ -0,0 +1,7 @@
+/// Encapsulates a command to toggle a post's seen status
+final class ReaderSeenAction {
+    func execute(with post: ReaderPost, context: NSManagedObjectContext, completion: (() -> Void)? = nil) {
+        let postService = ReaderPostService(managedObjectContext: context)
+        postService.toggleSeen(for: post, success: nil, failure: nil)
```

Comment:
Would it make sense to define a failure block to notify the user in the case of failure?

Reply:
Yea, I'll add a failure block.

Specifically, regarding `notify the user in the case of failure?` - it looks like the various failures in this menu are not consistent. At least one shows an alert, some others silently fail. I really didn't want to show an alert and disrupt the user just for this. I'm going to look at failures here a bit more and see if they all can show _something_ .